### PR TITLE
simx86: JumpGen: replace cond code with opc (opcode)

### PR DESCRIPTION
**Pull request by bartoldeman**
This allows the use of symbolic constants and merging some indentical fragments.

### CODE REVIEW

File: src/base/emu-i386/simx86/codegen-x86.c [hunk 1]
```diff
@@ -2150,7 +2150,7 @@ arith0:		{
 		}
 		else {
 			PopPushF(Cp);	// get flags from stack
-			G2M(0x70|cond,sz,Cp);	// normal cond
+			G2M(opc,sz,Cp);	// normal cond
```
Comment (stsp) [inline src/base/emu-i386/simx86/codegen-x86.c:2153]: Stalled comment.
Doesn't matter much, of course.

File: src/base/emu-i386/simx86/interp.c [hunk 2]
```diff
@@ -241,56 +241,48 @@ static unsigned int _JumpGen(unsigned int P2, int mode, int cond,
...
 	}
 	/* no break */
-	case 0x10:    /* uncond jmp */
+	case JMPsid: case JMPd:   /* uncond jmp */
 		if (dsp==0) {	// eb fe
 		    dbug_printf("!Forever loop!\n");
 		    leavedos_main(0xebfe);
 		}
-#if !defined(SINGLESTEP)
-		if (CONFIG_CPUSIM && !(EFLAGS & TF) && cond == 0x10) {
```
Comment (stsp) [inline src/base/emu-i386/simx86/interp.c:261]: Junk?
But its not change-logged.
OK though.

File: src/base/emu-i386/simx86/interp.c [hunk 3]
```diff
@@ -299,16 +291,16 @@ static unsigned int _JumpGen(unsigned int P2, int mode, int cond,
 #endif
 		}
 		if (CONFIG_CPUSIM)
```
Comment (stsp) [inline src/base/emu-i386/simx86/interp.c:293]: By the way, this condition can probably
be removed. As these things are accessed
by va_args, I don't think an extra arg will
make a problem for SIM.

File: src/base/emu-i386/simx86/interp.c [hunk 4]
```diff
@@ -2289,7 +2255,7 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 				Gen(S_DI, mode);
 				break;
 			case Ofs_SP:	/*4*/	 // JMP near indirect
-				PC = JumpGen(PC, mode, 0x40,
+				PC = JumpGen(PC, mode, JMPi,
```
Comment (stsp) [inline src/base/emu-i386/simx86/interp.c:2258]: But not opc? Why?

File: src/base/emu-i386/simx86/interp.c [hunk 5]
```diff
@@ -2856,7 +2824,7 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 			case 0x8e: /* JLEimmdisp */
 			case 0x8f: /* JNLEimmdisp */
 				{
-				  PC = JumpGen(PC, mode, (opc2-0x80),
+				  PC = JumpGen(PC, mode, JO+(opc2-0x80),
```
Comment (stsp) [inline src/base/emu-i386/simx86/interp.c:2827]: What does 0x80 stands for?
IMMEDbrm?

Comment (bartoldeman) [reply src/base/emu-i386/simx86/interp.c:2827]: `JOimmdisp`
I'll correct that later.

File: src/base/emu-i386/simx86/interp.c (hunk 4 above)
Comment (bartoldeman) [reply src/base/emu-i386/simx86/interp.c:2258]: opc is not sufficient here, as it's `0xff` (GRP2wrm) but were dealing with a 2 byte opcode of the form "ff /4" where "4" denotes the value of bits 3 - 5, so ff20 - ff27, ff60 - ff67, ffa0 - ffa7 and ffe0 - ffe7 in a disassembler. I defined JMPI as ff04 but could also use ff20 as some kind of "opcode base value", in that case one could use `(opc << 8) | (opc2 & 0x38)` ... would you like that?

**Comment by stsp** [comment]
Btw, can we get rid of that silly SKIPOP?
Its not safe with JIT, and doesn't do much
for SIM...

File: src/base/emu-i386/simx86/interp.c (hunk 2 above)
Comment (bartoldeman) [reply src/base/emu-i386/simx86/interp.c:261]: yes I left that in accidentally, should really be in a separate commit.

**Comment by bartoldeman** [comment]
I addressed your comments, but reverted the deletion of the "JMP:ignored" block as that should be a separate PR/commit.

**Comment by stsp** [comment]
Great cleanup, thanks!

**Pull request merged**